stdlib/csv.pyi

```diff
@@ -42,45 +46,50 @@ if sys.version_info >= (3,):
         lineterminator: str
         quoting: int
 
-if sys.version_info >= (3, 8):
-    _DRMapping = Dict[str, str]
-elif sys.version_info >= (3, 6):
-    _DRMapping = OrderedDict[str, str]
-else:
-    _DRMapping = Dict[str, str]
-
-class DictReader(Iterator[_DRMapping]):
+class DictReader(Generic[_T], Iterator[_DictReadMapping[_T, str]]):
+    fieldnames: Optional[Sequence[_T]]
     restkey: Optional[str]
     restval: Optional[str]
     reader: _reader
     dialect: _DialectLike
     line_num: int
-    fieldnames: Optional[Sequence[str]]
+    @overload
     def __init__(
         self,
         f: Iterable[Text],
+        fieldnames: Sequence[_T],
+        restkey: Optional[str] = ...,
+        restval: Optional[str] = ...,
+        dialect: _DialectLike = ...,
+        *args: Any,
+        **kwds: Any,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: DictReader[str],
+        f: Iterable[Text],
         fieldnames: Optional[Sequence[str]] = ...,
         restkey: Optional[str] = ...,
         restval: Optional[str] = ...,
         dialect: _DialectLike = ...,
         *args: Any,
         **kwds: Any,
     ) -> None: ...
-    def __iter__(self) -> DictReader: ...
+    def __iter__(self) -> DictReader[_T]: ...
     if sys.version_info >= (3,):
-        def __next__(self) -> _DRMapping: ...
+        def __next__(self) -> _DictReadMapping[_T, str]: ...
     else:
-        def next(self) -> _DRMapping: ...
+        def next(self) -> _DictReadMapping[_T, str]: ...
 
-class DictWriter(object):
-    fieldnames: Sequence[str]
+class DictWriter(Generic[_T]):
+    fieldnames: Sequence[_T]
     restval: Optional[Any]
     extrasaction: str
     writer: _writer
     def __init__(
         self,
         f: Any,
-        fieldnames: Iterable[str],
+        fieldnames: Iterable[_T],
```

Comment:
Just realized that `DictWriter` without `fieldnames` is a problem for this code

Reply:
Nevermind, you have to provide `fieldnames` when making a new `DictWriter`. But it should be `Sequence`, not `Iterable`, because it's iterated several times.